### PR TITLE
Add cURL extension

### PR DESCRIPTION
**Pull request by jonathangiardino**
## Description

Make HTTP requests directly from Raycast. Support JSON payload in markdown for POST, PUT and DELETE. 

- Main command is "Make HTTP request".
- Secondary command is "Requests History" where you can see a list of all the requests ran and you can either copy the cURL to the clipboard or run the request again.

## Screencast

https://user-images.githubusercontent.com/47124762/187899636-05ca0c31-6730-433e-97bc-07468c94f959.mp4

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by thomaspaulmann** [comment]
Looks like you pushed a submodule. Make sure the remove the git folder inside your extension before pushing it.

**Comment by jonathangiardino** [comment]
> Looks like you pushed a submodule. Make sure the remove the git folder inside your extension before pushing it.

Thanks, Thomas, that should be all right now.

**Comment by pernielsentikaer** [comment]
Hi @jonathangiardino 👋 

I hope you don't mind that I changed a few things, could you have a look? 🙂

**Comment by jonathangiardino** [comment]
> Hi @jonathangiardino 👋
> 
> I hope you don't mind that I changed a few things, could you have a look? 🙂

Of course not @pernielsentikaer, changes look great! Thanks!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

File: extensions/curl/src/requests.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  value: string;
+}
+
+interface RequestArg {
+  url: string;
+  payload: string;
+}
+
+export default function Requests() {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```
Comment (thomaspaulmann) [inline extensions/curl/src/requests.tsx:31]: It's better to set this to `true` initially to avoid flickering of the empty view.

Comment (jonathangiardino) [reply extensions/curl/src/requests.tsx:31]: Made the changes in my fork but the PR is closed. How do you usually manage these changes afterward? Open a new PR? @thomaspaulmann

Comment (thomaspaulmann) [reply extensions/curl/src/requests.tsx:31]: Yeah just another PR

**Pull request merged**